### DISCUSSION TITLE
Laggy /docs Page Help

### DISCUSSION THREAD

**Discussion by pandabull** [fastapi/fastapi#11097] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
#This probably isn't reproducible, but I add here for context 
@router.get("/exchange_chain", tags=["live data", "options"])
async def get_options_chain_by_exhange(chain: OptionsChainQuery = Depends()):
    """
    Retrieve standardized option chain
    - **currency**: required query parameter (from chain.currency)
    - **exchange**: required query parameter (from chain.exchange)
    """
    # here r is a redis client 
    data = await r.read_options_by_exchange(
        exchange=chain.exchange.value, currency=chain.currency.value.upper()
    )

    if data:
        resp = {
            "msg": "OK",
            "server_timestamp": now(),
            "data": data,
            "exchange": chain.exchange,
        }

        return ORJSONResponse(content=resp, media_type="application/json")
    else:
        raise HTTPException(500, detail="Internal server error")
```

### Description

Hello, I have recently created a FastAPI project, everything is working really well so far apart from the /docs page, I have the app running at 

[https://api.pandabull.io/docs](url)

Ideally I would like to keep this page for others to play around with the API , as it is quite nice especially for those of us unfamiliar with how APIs work etc. But the problem is when I use the **Try it out** button, let's take the /options/exchange_chain endpoint as an example. When I select the default parameters which results in a composed url of 

https://api.pandabull.io/options/exchange_chain?exchange=deribit&currency=BTC

I can run this no problem from the /docs page try it out, but after I run it once, then the entire page becomes unresponsive and I basically can't use it again unless I completely refresh the browser. 

## My Investigations 
Clearly this is something do to with the size of the response, because on other endpoints e.g. /options/futures_curve_by_exchange thing just work perfectly . I am also using Insomnia to test my endpoints. I tested the aforementioned composed url  (without accepting gzip) and the total response time was 300ms, and size is 357.5KB , it all renders nicely in the Insomnia UI. 

I also called it with curl locally with the following. 

`time curl "https://api.pandabull.io/options/exchange_chain?exchange=deribit&currency=BTC"
`

And again it takes like 300ms which is acceptable for me. 

I just can't seem to figure out what is causing the /docs page to become unresponsive. The same thing happens locally also, the page just basically jams up and won't work, sometimes even getting a pop-up saying 'unresponsive'. 

I haven't interfered with the docs page in any way, my app declaration looks as follows:

```
app = FastAPI(
    title="Pandabull.io",
    summary="Documentation for pandabull.io api methods",
    description="Hello this is the description section",
)
````

Any pointers would be much appreciated, keep in mind I am very new to this!

### Operating System

Linux

### Operating System Details

Ubuntu 23.10 both locally and on server

### FastAPI Version

0.109.0

### Pydantic Version

2.6.0

### Python Version

Python 3.11.6

**Comment by ahmedabdou14** [comment]
Hi, I tried the API. It works fine. It is slow because of the api is returning so much data.
I tested the API using Insomnia and it took 1 second to fetch the data. I would expect swagger to take a bit more time than that to build the UI components.

Regarding the page freezing, i did not experience any, did you push a fix by any chance?

**Comment by pandabull** [comment]
Hi, no fix yet. 

Not sure how it is working for you. I literally just checked again, and it runs the first time, albeit very slowly, but then if I want to run it again or run another method it lags really hard.

**Comment by jgould22** [comment, marked as answer]
This is more of a swagger doc issue I think.

Swagger has never handled large responses well and I have had the same experience.

Likely it is because it is trying to apply syntax highlighting using JS to a half megabyte of json in the response preview. 

Try disabling the syntax highlighting and see if that helps 

https://fastapi.tiangolo.com/how-to/configure-swagger-ui/#disable-syntax-highlighting

**Reply by pandabull** [reply]
Yes, this works thank you very much.
